src/compiler/factory.ts

```diff
@@ -1260,11 +1268,11 @@ namespace ts {
                     foundUseStrict = true;
                 }
 
-                target.push(source[i]);
+                target.unshift(source[i]);

```

Comment:
it will put the prologues as the first statement in the result array. I use unshift here because there is no guarantee in the order of this function when it gets call.